### PR TITLE
Even more debugger support

### PR DESCRIPTION
**Pull request by ben-clayton**
This PR contains 3 changes:

* Roll dxc, spirv-tools & swiftshader to latest
* Pass the `-fspv-debug=rich` flag to DXC when `DEBUG`-ing shaders. Add tests for this and other existing `DEBUG` parser logic.
* Fix up breakpoint handling for the new debugger.

### CODE REVIEW

**Comment by ben-clayton** [comment]
Failures (except for CI-check-format) appear to be caused by [b/169966461](https://issuetracker.google.com/issues/169966461)

File: src/amberscript/parser.cc [hunk 1]
```diff
@@ -2625,14 +2634,18 @@ Result Parser::ParseDebug() {
...
 
   return Result();
 }
 
-Result Parser::ParseDebugThread(debug::Events* dbg) {
+Result Parser::ParseDebugThread(debug::Events* dbg, Pipeline* pipeline) {
   auto token = tokenizer_->NextToken();
   if (token->AsString() == "GLOBAL_INVOCATION_ID") {
+    for (auto& shader : pipeline->GetShaders()) {
+      shader.SetEmitDebugInfo(true);
```
Comment (dj2) [inline src/amberscript/parser.cc:2646]: The other two places this is done guard on shader type, should this do the same?

Comment (ben-clayton) [reply src/amberscript/parser.cc:2646]: `GLOBAL_INVOCATION_ID` isn't tied to a specific shader type, so no.

**Comment by dj2** [comment]
lgtm once you get the build issues sorted out.

**Review by dj2 — APPROVED** [review]

**Pull request merged**